geomstats/geometry/general_linear.py

```diff
@@ -1,192 +1,125 @@
-        """Check if matrix belongs to GL(n).
+        Exponentiate a left-invariant vector field from a base point.
 
         Parameters
         ----------
-        mat
+        vector :        array-like, shape=[..., n, n]
+        base_point :    array-like, shape=[..., n, n]
+            Defaults to identity.
 
         Returns
         -------
-        belongs : bool
+        point :         array-like, shape=[..., n, n]
+            The left multiplication of `exp(vector)` with
+            `base_point`.
         """
-        mat = gs.to_ndarray(mat, to_ndim=3)
-
-        det = gs.linalg.det(mat)
-        belongs = ~gs.isclose(det, 0.)
-
-        belongs = gs.to_ndarray(belongs, to_ndim=1)
-        belongs = gs.to_ndarray(belongs, to_ndim=2, axis=1)
-
-        return belongs
-
-    def compose(self, mat_a, mat_b):
-        """Compose two matrices.
-
-        Parameters
-        ----------
-        mat_a
-        mat_b
+        expm = gs.linalg.expm
+        if base_point is None:
+            return expm(vector)
+        else:
+            return cls.mul(expm(vector), base_point)
 
-        Returns
-        -------
-        composition
+    @classmethod
+    def log(cls, point, base_point=None):
         """
-        mat_a = gs.to_ndarray(mat_a, to_ndim=3)
-        mat_b = gs.to_ndarray(mat_b, to_ndim=3)
-        composition = gs.einsum('nij,njk->nik', mat_a, mat_b)
-        return composition
-
-    def inverse(self, mat):
-        """Compute matrix inverse.
+        Search for a left-invariant vector field bringing base_point to point.
 
         Parameters
         ----------
-        mat
+        point :         array-like, shape=[..., n, n]
+        base_point :    array-like, shape=[..., n, n]
+            Defaults to identity.
 
         Returns
         -------
-        inverse
+        vector :        array-like, shape=[..., n, n]
+            A matrix such that `exp(vector, base_point) = point`.
         """
-        mat = gs.to_ndarray(mat, to_ndim=3)
-        return gs.linalg.inv(mat)
-
-    def exp_from_identity(self, tangent_vec, point_type=None):
-        """Compute group exponential at the identity.
-
-        Group exponential of the Lie group of
-        all invertible matrices at the identity.
-
-        Parameters
-        ----------
-        tangent_vec
-        point_type
-
-        Returns
-        -------
-        group_exp
-        """
-        tangent_vec = gs.to_ndarray(tangent_vec, to_ndim=3)
-        group_exp = gs.linalg.expm(tangent_vec)
-
-        return gs.real(group_exp)
+        logm = gs.linalg.logm
+        if base_point is None:
+            return logm(point)
+        else:
+            return logm(cls.mul(point, cls.inv(base_point)))
 
-    def exp_not_from_identity(
-            self, tangent_vec, base_point, point_type=None):
-        """Compute group exponential from a base point.
+    @classmethod
+    def orbit(cls, point, base_point=None):
+        r"""
```

Comment:
r""" ?

Reply:
flake8 told me to
(for the tex code and \ I think!)